### PR TITLE
feat: select predefined period on date range picker

### PR DESCRIPTION
**Pull request by larissa-kamily-brisa**
## Issue Number

#912 

## Description

It's  possible to select predefined period ranges
The definition of these periods is done dynamically, the developer chooses according to the needs of the project.

## Screenshots
[Gravação de tela de 2024-01-23 16-34-09.webm](https://github.com/Brisanet/ion/assets/138057627/8bc93c73-bcd9-469d-b0b1-e1f0c2106a0d)

## Compliance

- [x] I have verified that this change complies with our code and contribution policies.
- [x] I have verified that this change does not cause regressions and does not affect other parts of the code.

### CODE REVIEW

File: projects/ion/src/lib/picker/date-picker/date-picker.component.spec.ts [hunk 1]
```diff
@@ -95,4 +96,45 @@ describe('DatePickerCalendar', () => {
...
+      };
+      const spy = jest.spyOn(component, 'dateSelected');
+      component.rangePicker = true;
+
+      component.onSelectPredefinedRange(week);
+      fixture.detectChanges();
+      expect(spy).toHaveBeenCalled();
+      expect(component.currentDate).toEqual([todayText, sevenDaysAfter]);
+    });
+  });
```
Comment (deeved-hiuston-brisa) [inline projects/ion/src/lib/picker/date-picker/date-picker.component.spec.ts:139]: Implement tests for 15 and 30 day options.

**Comment by deeved-hiuston-brisa** [comment]
Good job! This feature will help us a lot.

File: projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts [hunk 2]
```diff
@@ -49,11 +52,11 @@ export class IonDatePickerCalendarComponent implements OnInit, OnChanges {
     }
   }
   @Input() calendarControlAction: CalendarControlActions;
+  @Input() selectedDay: Day[] = [];
```
Comment (iurynogueira) [inline projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts:55]:
```suggestion
  @Input() selectedDays: Day[] = [];
```

File: projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts [hunk 3]
```diff
@@ -65,11 +68,24 @@ export class IonDatePickerCalendarComponent implements OnInit, OnChanges {
...
+  nextMonth(): void {
+    this.calendar.goToNextMonth();
+  }
+
+  nextYear(): void {
+    this.calendar.goToNextYear(this.calendar.month.number - 1);
   }
 
   ngOnInit(): void {
+    arrangeDates(this.selectedDay);
```
Comment (iurynogueira) [inline projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts:88]:
```suggestion
    arrangeDates(this.selectedDay);
```

Check length this.selectedDay before

File: projects/ion/src/lib/picker/date-picker/date-picker.component.ts [hunk 4]
```diff
@@ -24,12 +31,14 @@ export class IonDatepickerComponent implements AfterViewInit {
...
     DEFAULT_INPUT_FORMAT;
   @Input() rangePicker: boolean;
+  @Input() predefinedRanges?: PreDefinedRangeConfig[] = [];
   @Output() event = new EventEmitter<string[]>();
   currentDate: string[];
   inputDate: string;
   showDatepicker = false;
   calendarMonth: string;
   calendarYear: string;
+  selectedDay: Day[] = [];
```
Comment (iurynogueira) [inline projects/ion/src/lib/picker/date-picker/date-picker.component.ts:41]:
```suggestion
  selectedDays: Day[] = [];
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]
Nice work @larissa-kamily-brisa

**Review by deeved-hiuston-brisa — APPROVED** [review]

**Review by wermeson-lopes-brisa — APPROVED** [review]

**Review by iurynogueira — APPROVED** [review]

**Pull request merged**